2-api/server.js

```diff
@@ -0,0 +1,227 @@
+});
+
+app.post("/customers/:customerId/orders", function (req, res) {
+  const customerId = req.params.customerId;
+  const orderDate = req.body.order_date;
+  const orderRef = req.body.order_reference;
+
+  pool
+    .query("SELECT * FROM customers WHERE id=$1", [customerId])
+    .then((result) => {
+      if (result.rowCount === 0) {
+        return res.status(400).send("Customer not Exist");
+      } else {
+        const query =
+          "INSERT INTO orders (order_date, order_reference, customer_id)VALUES ($1, $2, $3)";
+        pool
+          .query(query, [orderDate, orderRef, customerId])
+          .then(() => res.send("New order created"))
+          .catch((e) => console.error(e));
+      }
+    });
+});
+
+app.put("/customers/:customerId", function (req, res) {
+  const customerId = req.params.customerId;
+  const newName = req.body.name;
+  const newAdd = req.body.address;
+  const newCity = req.body.city;
+  const newCountry = req.body.country;
+
+  pool
+    .query("SELECT * FROM customers WHERE id=$1", [customerId])
+    .then((result) => {
+      if (result.rowCount === 0) {
+        return res.status(400).send("Customer not exist");
+      } else {
+        const query =
+          "UPDATE customers SET name=$1, address=$2, city=$3, country=$4 WHERE id=$5";
+
+        pool
+          .query(query, [newName, newAdd, newCity, newCountry, customerId])
+          .then(() => res.send("Customer info been update"))
+          .catch((e) => console.error(e));
+      }
+    });
+});
+
+app.delete("/orders/:orderId", function (req, res) {
+  const orderId = req.params.orderId;
+
+  pool
+    .query("DELETE FROM orders WHERE customer_id=$1", [orderId])
+    .then(() => {
+      pool
+        .query("DELETE FROM customers WHERE id=$1", [orderId])
+        .then(() => res.send("customer deleted"))
+        .catch((e) => console.error(e));
+    })
+    .catch((e) => console.error(e));
+});
+
+app.delete("/customers/:customerId", function (req, res) {
+  const customerId = req.params.customerId;
+
+  pool
+    .query("SELECT * FROM orders WHERE customer_id=$1", [customerId])
+    .then((result) => {
+      if (result.rowCount.length !== 0) {
+        res.status(400).send("Can't delete customer with orders");
+      } else {
+        pool
+          .query("DELETE FROM customers WHERE id=$1", [customerId])
+          .then(() => res.send("customer deleted"))
+          .catch((e) => console.error(e));
+      }
+    })
+    .catch((e) => console.error(e));
+});
+
+app.get("/customers/:customerId/orders", function (req, res) {
+  const customerId = req.params.customerId;
+
+  const query =
+    "SELECT o.order_reference, o.order_date, p.product_name, pa.unit_price, s.supplier_name, oi.quantity  FROM orders o INNER JOIN order_items oi ON o.id = oi.order_id INNER JOIN products p ON oi.product_id = p.id INNER JOIN product_availability pa ON oi.product_id = pa.prod_id INNER JOIN suppliers s ON oi.supplier_id = s.id WHERE customer_id=$1";
+
```

Comment:
You need to join order item with product availability on both the product id and the supplier id.